src/ansys/simai/core/utils/auth.py

```diff
@@ -42,75 +45,101 @@
 
 
 class _AuthTokens(BaseModel):
-    """Represents the OIDC tokens received from the auth server.
+    """Represents the OIDC tokens received from the auth server."""
 
-    The class can fetch and refresh these tokens.
-    It caches the tokens to disk automatically.
-    """
-
-    _EXPIRATION_BUFFER = timedelta(seconds=5)
+    # Time buffer so we treat nearly invalid tokens as invalid.
+    # Random so multiple processes don't refresh tokens at the same time.
+    EXPIRATION_BUFFER: ClassVar = random.randrange(5, 15)  # noqa: S311
 
-    cache_file: Optional[str] = None
     access_token: str
-    expires_in: int
-    refresh_expires_in: int
-    expiration: datetime = None
-    refresh_expiration: datetime = None
+    expiration: datetime
+    refresh_expiration: datetime
     refresh_token: str
-    # ...  (unused fields were removed for simplicity)
-
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        if self.expiration is None:
-            now = datetime.utcnow()
-            self.expiration = now + timedelta(seconds=self.expires_in)
-            self.refresh_expiration = now + timedelta(seconds=self.refresh_expires_in)
-        if self.cache_file is not None:
-            with open(self.cache_file, "w") as f:
-                f.write(self.model_dump_json())
+    # ...  (unused fields removed)
 
+    @model_validator(mode="before")
     @classmethod
-    def from_cache(cls, file_path: Optional[str]) -> "Optional[_AuthTokens]":
-        if file_path is None:
-            return None
+    def expires_in_to_datetime(cls, data: Any) -> dict:
+        assert isinstance(data, dict)
+        if "expiration" not in data:
+            # We want and store "expiration" but API responses contain "expires_in"
```

Comment:
We want __to__ store "expiration" ... ? The __and__ doesn't make much sense here